libs/martech/martech.js

```diff
@@ -97,49 +98,89 @@ const getDtmLib = (env) => ({
       : env.consumer?.marTechUrl || 'https://assets.adobedtm.com/d4d114c60e50/a0e989131fd5/launch-a27b33fc2dc0-development.min.js',
 });
 
-export default async function init({ persEnabled = false, persManifests }) {
+const setupEntitlementCallback = () => {
+  const setEntitlements = async (destinations) => {
+    const { default: parseEntitlements } = await import('../features/personalization/entitlements.js');
+    return parseEntitlements(destinations);
+  };
+
+  const getEntitlements = () => new Promise((resolve) => {
+    const handleEntitlements = (e) => {
+      if (e.detail?.result?.destinations?.length) {
+        resolve(setEntitlements(e.detail.result.destinations));
+      } else {
+        resolve([]);
+      }
+    };
+    window.addEventListener(ALLOY_SEND_EVENT, handleEntitlements, { once: true });
```

Comment:
A similar approach has been used in FEDS and noticed this event is fired with a small delay. Would it be possible to call `alloy('sendEvent')` directly and get the response? It might be faster, but I don't understand the exact use-case or limitations.

Reply:
I don't know yet, I think it's worth revisiting once this is in prod and we've checked perf